clients/cli/src/config.rs

```diff
@@ -0,0 +1,138 @@
+    },
+    spl_token_client::client::{ProgramClient, ProgramRpcClient, ProgramRpcClientSendTransaction},
+    std::{process::exit, rc::Rc, sync::Arc},
+};
+
+pub type Error = Box<dyn std::error::Error + Send + Sync>;
+
+pub fn println_display(config: &Config, message: String) {
+    match config.output_format {
+        OutputFormat::Display | OutputFormat::DisplayVerbose => {
+            println!("{}", message);
+        }
+        _ => {}
+    }
+}
+
+pub fn eprintln_display(config: &Config, message: String) {
+    match config.output_format {
+        OutputFormat::Display | OutputFormat::DisplayVerbose => {
+            eprintln!("{}", message);
+        }
+        _ => {}
+    }
+}
+
+pub struct Config {
+    pub rpc_client: Arc<RpcClient>,
+    pub program_client: Arc<dyn ProgramClient<ProgramRpcClientSendTransaction>>,
+    pub default_signer: Option<Arc<dyn Signer>>,
+    pub fee_payer: Option<Arc<dyn Signer>>,
+    pub output_format: OutputFormat,
+    pub dry_run: bool,
+    pub rent: Rent,
+}
+impl Config {
+    pub async fn new(
+        cli: Cli,
+        matches: ArgMatches,
+        wallet_manager: &mut Option<Rc<RemoteWalletManager>>,
+    ) -> Self {
+        // get the generic cli config struct
+        let cli_config = if let Some(config_file) = &cli.config_file {
+            solana_cli_config::Config::load(config_file).unwrap_or_else(|_| {
+                eprintln!("error: Could not load config file `{}`", config_file);
+                exit(1);
+            })
+        } else if let Some(config_file) = &*solana_cli_config::CONFIG_FILE {
+            solana_cli_config::Config::load(config_file).unwrap_or_default()
+        } else {
+            solana_cli_config::Config::default()
+        };
+
+        // create rpc client
+        let rpc_client = Arc::new(RpcClient::new_with_commitment(
+            cli.json_rpc_url.unwrap_or(cli_config.json_rpc_url),
+            CommitmentConfig::confirmed(),
+        ));
+
+        // and program client
+        let program_client = Arc::new(ProgramRpcClient::new(
+            rpc_client.clone(),
+            ProgramRpcClientSendTransaction,
+        ));
+
+        // resolve default signer
+        let default_path = cli_config.keypair_path;
+        let default_signer = signer_from_path(&matches, &default_path, "default", wallet_manager)
+            .ok()
+            .map(Arc::from);
+
+        // resolve fee-payer
+        let fee_payer = cli
+            .fee_payer
+            .and_then(|source| {
+                signer_from_source(&matches, &source, "fee_payer", wallet_manager).ok()
+            })
+            .map(Arc::from)
+            .or(default_signer.clone());
+
+        // determine output format
+        let output_format = match (cli.output_format, cli.verbose) {
+            (Some(json_format), _) => json_format,
+            (None, true) => OutputFormat::DisplayVerbose,
+            (None, false) => OutputFormat::Display,
+        };
+
+        // get rent
```

Comment:
apart from the rent field, this is entirely plagiarized from the single-pool program